src/js/smooch.jsx

```diff
@@ -196,6 +195,40 @@ export class Smooch {
         });
     }
 
+    monitorUrlChanges() {
+        const onhashchange = window.onhashchange;
+        const pushState = window.history.pushState;
+        const replaceState = window.history.replaceState;
+
+        window.onhashchange = function() {
```

Comment:
But that won't work based on the SO article you posted:

http://stackoverflow.com/questions/4570093/how-to-get-notified-about-changes-of-the-history-via-history-pushstate

5.5.9.1 Event definitions

"The popstate event is fired in certain cases when navigating to a session history entry.
According to this, there is no reason for popstate to be fired when you use pushState. But an event such as pushstate would come in handy. Because history is a host object, you should be careful with it, but Firefox seems to be nice in this case."